BUILD.md

```diff
@@ -1,66 +1,138 @@
-# WAF Build #
+# Building ArduPilot #
 
 Ardupilot is gradually moving from the make-based build system to
-[Waf](https://waf.io/).
+[Waf](https://waf.io/). You can read the [Waf Book](https://waf.io/book/) if
+you want to learn more about Waf.
 
-To keep access to Waf convenient, use the following alias from the
-root ardupilot directory:
+Waf should always be called from the ardupilot's root directory. Differently
+from the make-based build, with Waf there's a configure step to choose the
+board to be used (default is `sitl`).
 
-```bash
-alias waf="$PWD/modules/waf/waf-light"
-```
+## Basic usage ##
 
-You can also define the alias or create a function in your shell rc file (e.g.
-`~/.bashrc`).
+There are several commands in the build system for advanced usages, but here we
+list some basic and more used commands as example.
 
-You can read the [Waf Book](https://waf.io/book/) if you want to learn more
-about Waf.
+* **Build ArduCopter**
 
-## Calling waf ##
+    Here we use minlure as an example of Linux board. See more advanced usages
+    below to get a list of supported boards.
 
-Waf should always be called from the ardupilot's root directory.
+    ```sh
+    ./waf configure --board minlure
+    ./waf copter
+    ```
 
-Differently from the make-based build, with Waf there's a configure step
-to choose the board to be used (default is `sitl`):
+    The first command should be called only once or when you want to change a
+    configuration option. One configuration often used is the `--board` option to
+    switch from one board to another one. For example we could switch Pixhawk and
+    build again:
 
-```bash
-# Configure the Linux board
-waf configure --board=linux
-```
+    ```sh
+    ./waf configure --board px4-v2
+    ./waf copter
+    ```
 
-Waf build system is composed of commands. For example, the above command
-(`configure`) is for configuring the build. Consequently, in order to build, a
-"build" command is issued, thus `waf build`. That is the default command, so
-calling just `waf` is enough:
+* **List available boards**
 
-```bash
-# Build programs from bin group
-waf
 
-# Waf also accepts '-j' option to parallelize the build.
-waf -j8
-```
+    It's possible to get a list of supported boards on ArduPilot with the command
+    below
 
-To clean things up, use the `clean` or `distclean` command:
+    ```sh
+    ./waf list_boards
 
-```bash
-# Clean the build products, but keep configure information
-waf clean
+    ```
+
+* **Clean the build**
+
+    Commands `clean` and `distclean` can be used to clean the objects produced by
+    the build. The first keeps the configure information while the second cleans
+    everything.
+
+    Cleaning the build is very often not necessary and discouraged. We do
+    incremental builds reducing the build time by orders of magnitude.
+
+
+* **Upload or install**
+
+    Pixhawk board has and upload option so it uploads to the firmware built to
```

Comment:
I'd say that our build system provides the option `--upload`, which can be used to upload the firmware to a connected board, and that not all boards implement that. Then use Pixhawk as an example of one board that supports it.